pkg/adapter/quotas.go

```diff
@@ -108,10 +110,34 @@ type (
 
 	// QuotaResult2 provides return values from quota allocation calls on the handler
 	QuotaResult2 struct {
+		// The outcome status of the operation.
+		Status rpc.Status
+
 		// The amount of time until which the returned quota expires, this is 0 for non-expiring quotas.
 		ValidDuration time.Duration
 
 		// The total amount of quota returned, may be less than requested.
 		Amount int64
 	}
 )
+
+// GetStatus gets status embedded in the result.
+func (r QuotaResult2) GetStatus() rpc.Status { return r.Status }
+
+// SetStatus embeds status in result.
+func (r *QuotaResult2) SetStatus(s rpc.Status) { r.Status = s }
+
+// Combine combines other result with self.
+func (r *QuotaResult2) Combine(otherPtr interface{}) interface{} {
```

Comment:
Do we need to combine results simply in an array then ?
Or is this where we will enforce the single dispatch semantic ?